pkg/plugin/plugin.go

```diff
@@ -50,15 +51,44 @@ func (r *RpcPlugin) SetWeight(rollout *v1alpha1.Rollout, desiredWeight int32, ad
 			ErrorString: err.Error(),
 		}
 	}
+
 	if gatewayAPIConfig.HTTPRoute != "" {
-		return r.setHTTPRouteWeight(rollout, desiredWeight, additionalDestinations, &gatewayAPIConfig)
+		gatewayAPIConfig.HTTPRoutes = append(gatewayAPIConfig.HTTPRoutes, gatewayAPIConfig.HTTPRoute)
 	}
+
 	if gatewayAPIConfig.TCPRoute != "" {
-		return r.setTCPRouteWeight(rollout, desiredWeight, additionalDestinations, &gatewayAPIConfig)
+		gatewayAPIConfig.TCPRoutes = append(gatewayAPIConfig.TCPRoutes, gatewayAPIConfig.TCPRoute)
 	}
-	return pluginTypes.RpcError{
-		ErrorString: GatewayAPIManifestError,
+
+	if len(gatewayAPIConfig.HTTPRoutes) == 0 && len(gatewayAPIConfig.TCPRoutes) == 0 {
+		return pluginTypes.RpcError{
+			ErrorString: GatewayAPIManifestError,
+		}
 	}
+
+	for _, name := range gatewayAPIConfig.HTTPRoutes {
+		err := r.setHTTPRouteWeight(rollout, desiredWeight, additionalDestinations, GatewayAPITrafficRouting{
+			Namespace: gatewayAPIConfig.Namespace,
+			HTTPRoute: name,
+		})
+		if err.HasError() {
+			return pluginTypes.RpcError{
+				ErrorString: fmt.Sprintf("set weight for HTTPRoute %q: %s", name, err),
+			}
+		}
+	}
```

Comment:
Have code duplication. Lets make seperate function for that
```go
forEachGatewayAPIRoute(listOfRoutes, func (name) error {
     return r.setHTTPRouteWeight(rollout, desiredWeight, additionalDestinations, GatewayAPITrafficRouting{
			Namespace: gatewayAPIConfig.Namespace,
			HTTPRoute: name,
    })
})
```
And below you can reuse it

Reply:
Looks overkill, but I guess whatever it takes to get this done ;)

Reply:
For example one of the feature that will be released on this week
https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-gatewayapi/pull/31

Reply:
And Im sure it will grow )